src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +900,208 @@ public function in_default_dir( $src ) {
 		return false;
 	}
 
+	/**
+	 * This overrides the add_data method from WP_Dependencies, to support normalizing of $args.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'strategy' === $key ) {
+			if ( ! empty( $value ) && ! $this->is_delayed_strategy( $value ) ) {
+				_doing_it_wrong(
+					__METHOD__,
+					sprintf(
+						/* translators: 1: $strategy, 2: $handle */
+						__( 'Invalid strategy `%1$s` defined for `%2$s` during script registration.' ),
+						$value,
+						$handle
+					),
+					'6.3.0'
+				);
+				return false;
+			} elseif ( empty( $this->registered[ $handle ]->src ) && $this->is_delayed_strategy( $value ) ) {
+				_doing_it_wrong(
+					__METHOD__,
+					sprintf(
+						/* translators: 1: $strategy, 2: $handle */
+						__( 'Cannot supply a strategy `%1$s` for script `%2$s` because it does not have a `src` value.' ),
+						$value,
+						$handle
+					),
+					'6.3.0'
+				);
+				return false;
+			}
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Checks all handles for any delayed inline scripts.
+	 *
+	 * @since 6.3.0
+	 * @see WP_Scripts::should_delay_inline_script()
+	 *
+	 * @param string[] $handles Handles to check.
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script( array $handles ) {
+		foreach ( $handles as $handle ) {
+			foreach ( array( 'before', 'after' ) as $position ) {
+				if (
+					$this->get_data( $handle, $position ) &&
+					$this->should_delay_inline_script( $handle, $position )
+				) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Gets all dependents of a script.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle The script handle.
+	 * @return string[] Script handles.
+	 */
+	private function get_dependents( $handle ) {
+		// Check if dependents map for the handle in question is present. If so, use it.
+		if ( array_key_exists( $handle, $this->dependents_map ) ) {
```

Comment:
As a rule, `array_key_exists` is slower than `isset`. See https://stackoverflow.com/questions/6337893/why-is-array-key-exists-1000x-slower-than-isset-on-referenced-arrays.

Reply:
OK. I'll change. Still, `isset()` can be bad for the same reasons as `empty()` in https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/

Reply:
Done in https://github.com/10up/wordpress-develop/pull/80

Reply:
FWIW I think `isset()` is perfectly reasonable to use, and faster than `array_key_exists()`. Looking at the article, I don't see it mentioning problems with using `isset()`, it's only about `empty()`. Yes, `isset( $some_var )` is pointless, but `isset( $arr[ $key ] )` is not.

Reply:
The problem with both `empty()` and `isset()` is that they mask issues related to typos. If I were to do this due to a hypothetical sticky key on my keyboard:

```php
if ( isset( $this->dependents_mappppppppppp[ $handle ] ) ) {
```

No error will ensue, but it will always be false. This, however, would be caught by using something else like `array_key_exists()`. 

In any case, static analysis will catch it. So I don't feel strongly about this.